bundles/org.openhab.binding.haassohnpelletstove/src/main/java/org/openhab/binding/haassohnpelletstove/internal/HaasSohnpelletstoveHandler.java

```diff
@@ -0,0 +1,327 @@
+        boolean validConfig = true;
+        String errors = "";
+        String statusDescr = null;
+        if (config.refreshRate < 0 && config.refreshRate > 999) {
+            errors += " Parameter 'refresh Rate' greater then 0 and less then 1000.";
+            statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1000.";
+            validConfig = false;
+        }
+        if (config.hostIP == null) {
+            errors += " Parameter 'hostIP' must be configured.";
+            statusDescr = "IP Address must be configured!";
+            validConfig = false;
+        }
+        if (config.hostPIN == null) {
+            errors += " Parameter 'hostPin' must be configured.";
+            statusDescr = "PIN must be configured!";
+            validConfig = false;
+        }
+        errors = errors.trim();
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            if (serviceCommunication.refreshOvenConnection(message, this.getThing().getUID().toString())) {
+                if (updateOvenData(null)) {
+                    updateStatus(ThingStatus.ONLINE);
+                    updateLinkedChannels();
+                }
+            } else {
+                logger.info("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+            }
+        } else {
+            logger.info("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    private void updateLinkedChannels() {
+        verifyLinkedChannel(CHANNELISTEMP);
+        verifyLinkedChannel(CHANNELMODE);
+        verifyLinkedChannel(CHANNELPOWER);
+        verifyLinkedChannel(CHANNELSPTEMP);
+        verifyLinkedChannel(CHANNELECOMODE);
+        verifyLinkedChannel(CHANNELIGNITIONS);
+        verifyLinkedChannel(CHANNELMAINTENANCEIN);
+        verifyLinkedChannel(CHANNELCLEANINGIN);
+        verifyLinkedChannel(CHANNELCONSUMPTION);
+        verifyLinkedChannel(CHANNELONTIME);
+        if (!linkedChannels.isEmpty()) {
+            logger.info("Start automatic refreshing");
+            updateOvenData(null);
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+    }
+
+    private void verifyLinkedChannel(String channelID) {
+        if (isLinked(channelID) && !linkedChannels.containsKey(channelID)) {
+            linkedChannels.put(channelID, true);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.info("Disposing Haas and Sohn Pellet stove handler.");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
ok